### ISSUE TITLE
How to do Visual embedding

### ISSUE THREAD

**Issue by WatershedXiaolan** [WatershedXiaolan/Xiaolan-s-Gitbook#358]
### Auto-encoders
Auto-encoders use neural networks consisting of both an encoder and a decoder. They first learn to compress the raw image pixel data to a small dimension via an encoder model and then try to de-compress it via a decoder to re-generate the same input image. **The last layer of encoder determines the dimension of the embedding**, which should be sufficiently large to capture enough information about the image so that the decoder can decode it.
<img width="814" alt="Screen Shot 2022-03-31 at 8 52 47 PM" src="https://user-images.githubusercontent.com/5552371/161179237-c89c9c1a-8ac7-48ae-879c-86e77cf0fc0b.png">

**Comment by WatershedXiaolan** [comment]
###Visual supervised learning tasks[#](https://www.educative.io/courses/grokking-the-machine-learning-interview/B8xpO2XGxXo#Visual-supervised-learning-tasks)

Visual supervised learning tasks such as image classification or object detection, are generally set up as convolution layers, pooling layers, and fully connected network layers, followed by final classification(softmax) layers. Let’s consider the example of the **ImageNet VGG16** model that is shown in the figure below. The input passes through a set of convolution, pooling, and fully connected layers to the last softmax layer for the final classification task. The penultimate layer before softmax captures all image information in a vector such that it can be used to classify the image correctly. So, **we can use the penultimate layer value of a pre-trained model as our image embedding.**

<img width="878" alt="Screen Shot 2022-03-31 at 8 58 47 PM" src="https://user-images.githubusercontent.com/5552371/161179779-ca6fa036-21ab-4456-8ee3-29232a83e62e.png">
